src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/ResponseEventsAboutToSubmit.java

```diff
@@ -82,17 +87,77 @@ private void validateChangedCaseCode(PreSubmitCallbackResponse<SscsCaseData> res
         final Optional<CaseDetails<SscsCaseData>> caseDetailsBefore = callback.getCaseDetailsBefore();
 
         if (response.getData().getAppeal().getBenefitType() != null
-            && Benefit.CHILD_SUPPORT.getShortName().equalsIgnoreCase(response.getData().getAppeal().getBenefitType().getCode())
-            && !Benefit.CHILD_SUPPORT.getCaseLoaderKeyId().contains(response.getData().getBenefitCode())
             && caseDetailsBefore.isPresent()
             && caseDetailsBefore.get().getCaseData().getBenefitCode() != null
-            && !caseDetailsBefore.get().getCaseData().getBenefitCode().equals(callback.getCaseDetails().getCaseData().getBenefitCode())) {
+            && !caseDetailsBefore.get().getCaseData().getBenefitCode()
+            .equals(callback.getCaseDetails().getCaseData().getBenefitCode())) {
 
-            if (response.getData().getOtherParties() != null && response.getData().getOtherParties().size() > 0) {
-                response.addError("Benefit code cannot be changed on cases with registered 'Other Party'");
-            } else {
-                response.addWarning("The benefit code will be changed to a non-child support benefit code");
+            if (Benefit.CHILD_SUPPORT.getShortName()
+                .equalsIgnoreCase(response.getData().getAppeal().getBenefitType().getCode())
+                && !Benefit.CHILD_SUPPORT.getCaseLoaderKeyId().contains(response.getData().getBenefitCode())) {
+                if (response.getData().getOtherParties() != null && response.getData().getOtherParties().size() > 0) {
+                    response.addError("Benefit code cannot be changed on cases with registered 'Other Party'");
+                } else {
+                    response.addWarning("The benefit code will be changed to a non-child support benefit code");
+                }
+            }
+
+            if ((isSscs5Case(response.getData().getAppeal().getBenefitType().getCode())
+                && !buildSscs5BenefitCaseLoaderKeyId().contains(response.getData().getBenefitCode()))
+                || (!isSscs5Case(response.getData().getAppeal().getBenefitType().getCode())
+                && buildSscs5BenefitCaseLoaderKeyId().contains(response.getData().getBenefitCode()))) {
+                response.addError("Benefit code cannot be changed to the selected code");
+            }
+        }
+    }
+
+    private boolean isSscs5Case(String benefitName) {
+        return benefitName != null ? Benefit.TAX_CREDIT.getShortName().equals(benefitName)
```

Comment:
No need for this method. Can change the calling method to something like `if SscsType.Sscs5.equals(Benefit.findBenefitByShortName(response.getData().getAppeal().getBenefitType().getCode()).getSscsType())`

Reply:
I have modified the method isSscs5Case